test_app/main.py

```diff
@@ -4,13 +4,24 @@
 
 import strawberry
 import uvicorn
+from api.types.sample import SampleCreateInput
 from platformics.api.setup import get_app, get_strawberry_config
 from platformics.api.core.error_handler import HandleErrors
 from platformics.settings import APISettings
 from database import models
+from platformics.api.core.strawberry_extensions import register_plugin
 
 from api.mutations import Mutation
 from api.queries import Query
+from typing import Any
+from strawberry.types import Info
+
+
+@register_plugin("before", "sample", "create")
+def validate_sample_name(source: Any, info: Info, **kwargs: SampleCreateInput) -> None:
```

Comment:
I think `**kwargs` might include everything that gets sent to the resolver, so in addition to `SampleCreateInput`, `session`, `cerbos_client`, `principal`, `is_system_user`, etc.